### DISCUSSION TITLE
Use of the Porous Flow module with a 2-phase and 3 fluid components system (no chemistry)

### DISCUSSION THREAD

**Discussion by alessandrovacbz** [idaholab/moose#23015] [Q&A Modules: General] [answered]
Hi all,
I need to simulate a porous flow consisting of:
> 2 phases: water(liquid) + gas
> 3 fluid components: water (liquid/vapour) + H2 + CO2
with no chemistry.
I wasn't able to find a similar case in the many input file examples.
I tried with "PorousFlowWaterNCG" and "PorousFlow2PhasePS" but I have problems in specifiying and using simultaneously the properties of the three fluids (it seems that I can only specify on a per phase base, not on per single component).
Can anyone help me, please, in addressing this case?
Thank you.
Best regards,
Alessandro

**Comment by cpgr** [comment, marked as answer]
Yeah, while this is technically possible the actual bits of code to enable it aren't in the publicly available MOOSE repository yet. There are two levels of complexity - one with no gas dissolution into the liquid phase, and one allowing gas to dissolve. The former is a lot easier!

I can help you out in the new year when I'm back at work if you like - just send me a reminder as I'll probably forget!

**Reply by alessandrovacbz** [reply]
Thank you for the reply!
I will send you the reminder.
Best regards,
Alessandro

**Comment by alessandrovacbz** [comment]
Hi Chris,
excuse me for bothering you. I'm writing this email to remind you
(according to your past suggestion) about the possibility of using the
Porous Flow module with a 2-phase and many fluid components (3 or more)
system (no chemistry).
I tried the easy way, defining an "IdealRealGasMixtureFluidProperties"
class in the "Fluid Properties" block and using it in the "Materials" with
the "PorousFlow2PhasePS" flow model.
It didn't work, probably for an incorrect class definition.
Could you help me in solving the problem?
Can I send you the *.i file?
In perspective, I suppose that the use of the "PorousFlowWaterNCG" flow
model would be the better solution, but it is also the harder one
to implement for more than one gaseous component.
Thank you.
Best regards,
Alessandro




Il giorno ven 23 dic 2022 alle ore 22:37 Chris Green <
***@***.***> ha scritto:

**Reply by garciapintado** [reply]
Hi Alessandro,
Just to mention that PorousFlowWaterNCG is if you need to allow gases to dissolve in the liquid phase (the harder option mentioned by @cpgr), which likely should be extended for multi-component situations. Although likely you already know that...
Actually, if this is your case, I am also curious to see how this would work, but haven't managed to do it myself by now

**Comment by alessandrovacbz** [comment]
Hi Chris, Hi Javier,

In fact, I'm waiting for some useful hints from Chris regarding the way on
how to modify MOOSE code to describe multi-component situations, due to
their implementation difficulties.

Best regards,

Alessandro


Il giorno mar 17 gen 2023 alle ore 15:56 Javier Garcia-Pintado <
***@***.***> ha scritto:

**Reply by cpgr** [reply]
Hi Alessandro,

I'm sort of back from holidays now so can try and help you out. Did you want to have gas dissolving in the water? I haven't looked at the code I have for this for a while, so it might need a little bit of work to make sure that it compiles and runs.

**Reply by sinaomrani96** [reply]
Hello everybody,

Since I faced similar issue #23920, I was wondering if this thread led to a solution. I appreciate it if you could share the solution if possible or give me a hint as well. Thanks.

**Comment by alessandrovacbz** [comment]
Hi Chris,

Thank you. Effectively, having gas dissolution would be the best choice for
me.

Best,
Alessandro


Il giorno gio 19 gen 2023 alle ore 01:30 Chris Green <
***@***.***> ha scritto:
